packages/core/src/xml-schema/xml-schema-model-adapter.ts

```diff
@@ -236,6 +237,15 @@ class XmlSchemaAdapter {
     return this.model.psmIri !== classData.structureSchema;
   }
 
+  currentPath(): string {
+    for (const artifact of this.specification.artefacts) {
+      if (artifact.generator === XML_SCHEMA.Generator) {
+        return artifact.publicUrl;
+      }
+    }
+    return "";
+  }
```

Comment:
You should use the artifact object provided to XmlSchemaGenerator class instead of searching it. In general, it should be possible to generate multiple artifacts from the same generator but with different options. Therefore searching only by a generator type is not sufficient.